src/Features/CSharp/Portable/Completion/KeywordRecommenders/RefKeywordRecommender.cs

```diff
@@ -85,7 +87,35 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
             syntaxTree.IsPossibleLambdaParameterModifierContext(position, context.LeftToken, cancellationToken) ||
             context.TargetToken.IsConstructorOrMethodParameterArgumentContext() ||
             context.TargetToken.IsXmlCrefParameterModifierContext() ||
-            IsValidNewByRefContext(syntaxTree, position, context, cancellationToken);
+            IsValidNewByRefContext(syntaxTree, position, context, cancellationToken) ||
+            IsAllowsRefStructConstraintContext(context);
+    }
+
+    private static bool IsAllowsRefStructConstraintContext(CSharpSyntaxContext context)
+    {
+        // cases:
+        //    where T : allows |
+        //    where T : struct, allows |
+        //    where T : new(), allows |
+        //    where T : Goo, allows |
+        // note: 'allows ref struct' can't come after a 'class' constraint.
+
+        var token = context.TargetToken;
+
+        if (token.Kind() == SyntaxKind.IdentifierToken && SyntaxFacts.GetContextualKeywordKind((string)token.Value!) == SyntaxKind.AllowsKeyword &&
+            token.Parent is IdentifierNameSyntax identifier && identifier.Identifier == token &&
+            identifier.Parent is TypeConstraintSyntax typeConstraint && typeConstraint.Type == identifier &&
+            typeConstraint.Parent is TypeParameterConstraintClauseSyntax constraintClause)
+        {
+            if (!constraintClause.Constraints
+                    .OfType<ClassOrStructConstraintSyntax>()
+                    .Any(c => c.ClassOrStructKeyword.Kind() == SyntaxKind.ClassKeyword))
```

Comment:
we'd be fine with this being dropped.

Reply:
> we'd be fine with this being dropped.  

Following what we do for `new` constraint